<launch>

  <arg name="odom_topic" value="/ugv/odometry" />
  <arg name="traj_topic" value="/planner/trajectory" />
  <arg name="cmd_topic" value="/ugv/command" />
  <arg name="sensor_topic" value="/ugv/laser" />
  <arg name="mesh_topic" value="/ugv/mesh" />
  <arg name="map_topic" value="/global_map" />

  <rosparam command="load" file="$(find mpc_controller)/params/controller.yaml" />
  <rosparam command="load" file="$(find kinematics_simulator)/params/simulator.yaml" />
  <rosparam command="load" file="$(find random_map_generator)/params/map.yaml" />

  <node pkg="random_map_generator" type="egeneration_node" name="egeneration_node" output="screen" required="true">
    <rosparam file="$(find plan_manage)/config/planning.yaml" command="load" />
    <param name="map/map_id" value="30000" type="int"/>
    <remap from ="~odom" to="$(arg odom_topic)"/>
    <remap from="~local_cloud" to="$(arg sensor_topic)"/>
    <remap from="~global_cloud" to="$(arg map_topic)"/> 
  </node>


    <!-- <node pkg="random_map_generator" type="egeneration_node" name="egeneration_node2" output="screen" required="true">
    <rosparam file="$(find plan_manage)/config/planning.yaml" command="load" />
    <param name="map/map_id" value="40000" type="int"/>
    <remap from ="~odom" to="$(arg odom_topic)"/>
    <remap from="~local_cloud" to="$(arg sensor_topic)"/>
    <remap from="~global_cloud" to="$(arg map_topic)"/> 
  </node>

    <node pkg="random_map_generator" type="egeneration_node" name="egeneration_node3" output="screen" required="true">
    <rosparam file="$(find plan_manage)/config/planning.yaml" command="load" />
    <param name="map/map_id" value="50000" type="int"/>
    <remap from ="~odom" to="$(arg odom_topic)"/>
    <remap from="~local_cloud" to="$(arg sensor_topic)"/>
    <remap from="~global_cloud" to="$(arg map_topic)"/> 
  </node>

    <node pkg="random_map_generator" type="egeneration_node" name="egeneration_node4" output="screen" required="true">
    <rosparam file="$(find plan_manage)/config/planning.yaml" command="load" />
    <param name="map/map_id" value="60000" type="int"/>
    <remap from ="~odom" to="$(arg odom_topic)"/>
    <remap from="~local_cloud" to="$(arg sensor_topic)"/>
    <remap from="~global_cloud" to="$(arg map_topic)"/> 
  </node>

    <node pkg="random_map_generator" type="egeneration_node" name="egeneration_node5" output="screen" required="true">
    <rosparam file="$(find plan_manage)/config/planning.yaml" command="load" />
    <param name="map/map_id" value="70000" type="int"/>
    <remap from ="~odom" to="$(arg odom_topic)"/>
    <remap from="~local_cloud" to="$(arg sensor_topic)"/>
    <remap from="~global_cloud" to="$(arg map_topic)"/> 
  </node>

    <node pkg="random_map_generator" type="egeneration_node" name="egeneration_node6" output="screen" required="true">
    <rosparam file="$(find plan_manage)/config/planning.yaml" command="load" />
    <param name="map/map_id" value="80000" type="int"/>
    <remap from ="~odom" to="$(arg odom_topic)"/>
    <remap from="~local_cloud" to="$(arg sensor_topic)"/>
    <remap from="~global_cloud" to="$(arg map_topic)"/> 
  </node>

    <node pkg="random_map_generator" type="egeneration_node" name="egeneration_node7" output="screen" required="true">
    <rosparam file="$(find plan_manage)/config/planning.yaml" command="load" />
    <param name="map/map_id" value="90000" type="int"/>
    <remap from ="~odom" to="$(arg odom_topic)"/>
    <remap from="~local_cloud" to="$(arg sensor_topic)"/>
    <remap from="~global_cloud" to="$(arg map_topic)"/> 
  </node>

    <node pkg="random_map_generator" type="egeneration_node" name="egeneration_node8" output="screen" required="true">
    <rosparam file="$(find plan_manage)/config/planning.yaml" command="load" />
    <param name="map/map_id" value="100000" type="int"/>
    <remap from ="~odom" to="$(arg odom_topic)"/>
    <remap from="~local_cloud" to="$(arg sensor_topic)"/>
    <remap from="~global_cloud" to="$(arg map_topic)"/> 
  </node>
   -->
  <!-- rviz -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find random_map_generator)/rviz/default.rviz" required="true" />

</launch>